src/compile/data/parse.ts

```diff
@@ -275,11 +273,13 @@ export function parseData(model: Model): DataComponent {
       stack.parent = head;
       head = stack;
     }
+  }
 
-    const nonPosFilter = NonPositiveFilterNode.make(model);
-    if (nonPosFilter) {
-      nonPosFilter.parent = head;
-      head = nonPosFilter;
+  if (isUnitModel(model) || isFacetModel(model)) {
```

Comment:
row / column channel in facet never have scales, so we don't really need to include `|| isFacetModel(model)` here.